app/controllers/admin/articles_controller.rb

```diff
@@ -29,12 +29,20 @@ def update
       redirect_to admin_articles_path, notice: 'Premium article approved for publication'  
     elsif @article.rejected?
       redirect_to admin_articles_path, notice: 'Article not approved for publication'
-    elsif @article.for_revision?
+    elsif @article.for_revision? && URI(request.referer).path == admin_articles_path
       @article.update_attribute(:comment, article_params[:comment])
       redirect_to admin_articles_path, notice: "Article not approved for publication, please see comments: #{@article.comment}"
+    elsif @article.for_revision? && URI(request.referer).path == edit_admin_article_path
+      @article.update(article_params)
+      @article.update_attribute(:status, :pending)
+      redirect_to admin_root_path, notice: "Your article was successfully re-submitted"
     end
   end
 
+  def edit
+    @article = Article.find(params[:id])
```

Comment:
Place this in a `before_action` You have repeating code in the `update` method/action.

Reply:
ah yes - that makes sense